spec/support/handler/profileHandlerspec.js

```diff
@@ -0,0 +1,148 @@
+            var res = await ph.UpdateProfile(null, profileID);
+            expect(res.getCode()).toBe(400);
+            var res = await ph.UpdateProfile(userID, null);
+            expect(res.getCode()).toBe(400);
+           
+        });
+    });
+
+    describe("createProfile method", () => {
+        let ph = new ProfileHandler();
+      
+        beforeEach(() => {
+          userID = generateShortUUID();
+        });
+      
+        it("Req body, username, createdAt, age, bmi, height, weight existence check", async () => {
+          ph.postValue = '{"height": 175, "weight": 70, "bmi": 22.86}';
+          var res = await ph.createProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "height": 175, "weight": 70}';
+          var res = await ph.createProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": 22.86}';
+          var res = await ph.createProfile(userID);
+          expect(res.getCode()).toBe(400);
+        });
+      
+        it("Req body, username, createdAt, age, bmi, height, weight enum", async () => {
+          ph.postValue = '{"username": null, "createdAt": "2023-01-01", "age": 30, "bmi": 22.86, "height": 175, "weight": 70}';
+          var res = await ph.createProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "invalid_date", "age": 30, "bmi": 22.86, "height": 175, "weight": 70}';
+          var res = await ph.createProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": "invalid_age", "bmi": 22.86, "height": 175, "weight": 70}';
+          var res = await ph.createProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": "invalid_bmi", "height": 175, "weight": 70}';
+          var res = await ph.createProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe","createdAt": "2023-01-01","age": 30,"bmi": 22.86, "height": "invalid_height","weight": 70}';
+          var res = await ph.createProfile(userID); 
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": 22.86, "height": 175, "weight": "invalid_weight"}';
+          var res = await ph.createProfile(userID); // Invalid weight format
+          expect(res.getCode()).toBe(400);
+        });
+      
+        //it("Create profile successfully", async () => {
+          //ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": 22.86, "height": 175, "weight": 70}';
```

Comment:
wonder why this is commented out shouldnt it work how else can we test if its successful